.github/workflows/perf-test.yml

```diff
@@ -0,0 +1,158 @@
+name: Performance Test
```

Comment:
I don't understand why we want this? Shouldn't this be what the test-plans perf runner does? The bash script here is not great (there's scripting inside scripting!). I don't think this is maintainable (I sure don't want to maintain this). We should have a single source of truth for perf – test test-plans perf runner.

I think the desire here is that we'll be able to "catch regressions in CI". That's very hard to do, and probably impossible on shared GitHub Actions runner and in JS. Realistically the path forward to improving performance looks like this:

1. We record a baseline performance in a reproducible and realistic environment. This is @mxinden's work in the test-plans perf runner.
2. We run a profiler to identify bottlenecks. Or intuitively identify bottlenecks.
3. Fix the bottlenecks.
4. Run the test from step 1.

To protect from regressions, for now I suggest we do this on each release rather than per PR.

Reply:
Good point @MarcoPolo and as @achingbrain mentioned there are resource constraints as well.

I will just remove this workflow for now then post-merge of https://github.com/libp2p/test-plans/pull/244 we can introduce a workflow to run on release / main in a follow up PR.